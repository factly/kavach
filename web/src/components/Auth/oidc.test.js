import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { act } from '@testing-library/react';
import { mount } from 'enzyme';
import '../../matchMedia.mock';
import OIDC from './oidc';

const config = {
  action:
    'http://127.0.0.1:4455/.ory/kratos/public/self-service/methods/oidc/auth/3b21e9d8-1b0a-4e12-8eb0-d77fdeede28e',
  method: 'POST',
  fields: [
    {
      name: 'csrf_token',
      type: 'hidden',
      required: true,
      value:
        'D5sFfeYIa9TxEij3vANoLDa9OIFip2jLmsF0O6lc/m50YjviU1/75iZPRP2WL6R94MOIXA+9mjlHj1MbVcpX+w==',
    },
    {
      name: 'provider',
      type: 'submit',
      value: 'github',
    },
    {
      name: 'provider',
      type: 'submit',
      value: 'google',
    },
  ],
};
describe('OIDC component', () => {
  describe('snapshot testing', () => {
    it('should render the component', () => {
      let component;
      act(() => {
        component = mount(<OIDC config={config} />);
      });
      expect(component).toMatchSnapshot();
    });

    it('should render the component when flow and loginmethod are not default', () => {
      let component;
      act(() => {
        component = mount(<OIDC config={config} flow="registration" loginMethod="oidc" />);
      });
      expect(component).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    it('renders a Github button when the UI object contains a Github node', () => {
      const ui = {
        action: '/',
        method: 'POST',
        nodes: [{ group: 'oidc', attributes: { value: 'github' } }]
      };
      render(<OIDC ui={ui} />);
      const githubButton = screen.getByText(/continue with github/i);
      expect(githubButton).toBeInTheDocument();
    });
    it('renders a Google button when the UI object contains a Google node', () => {
      const ui = {
        action: '/',
        method: 'POST',
        nodes: [{ group: 'oidc', attributes: { value: 'google' } }]
      };
      render(<OIDC ui={ui} />);
      const googleButton = screen.getByText(/continue with google/i);
      expect(googleButton).toBeInTheDocument();
    });
    it('calls withOIDC with the value "github" when the Github button is clicked', () => {
      const ui = {
        action: '/',
        method: 'POST',
        nodes: [{ group: 'oidc', attributes: { value: 'github', name: "csrf_token" } },]
      };
      const submitSpy = jest.spyOn(window.HTMLFormElement.prototype, 'submit');

      const wrapper = mount(<OIDC ui={ui} />);
      const githubButton = wrapper.find('button').at(0);
      githubButton.simulate('click');
      expect(submitSpy).toHaveBeenCalled();
      submitSpy.mockRestore();
    });
    it('calls withOIDC with the value "google" when the Google button is clicked', () => {
      const ui = {
        action: '/',
        method: 'POST',
        nodes: [{ group: 'oidc', attributes: { value: 'google', name: "csrf_token" } },]
      };
      const submitSpy = jest.spyOn(window.HTMLFormElement.prototype, 'submit');

      const wrapper = mount(<OIDC ui={ui} />);
      const githubButton = wrapper.find('button').at(0);
      githubButton.simulate('click');
      expect(submitSpy).toHaveBeenCalled();
      submitSpy.mockRestore();
    });
  });
});
