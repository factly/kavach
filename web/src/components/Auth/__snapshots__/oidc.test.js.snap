// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OIDC component snapshot testing should render the component 1`] = `
<OIDC
  config={
    Object {
      "action": "http://127.0.0.1:4455/.ory/kratos/public/self-service/methods/oidc/auth/3b21e9d8-1b0a-4e12-8eb0-d77fdeede28e",
      "fields": Array [
        Object {
          "name": "csrf_token",
          "required": true,
          "type": "hidden",
          "value": "D5sFfeYIa9TxEij3vANoLDa9OIFip2jLmsF0O6lc/m50YjviU1/75iZPRP2WL6R94MOIXA+9mjlHj1MbVcpX+w==",
        },
        Object {
          "name": "provider",
          "type": "submit",
          "value": "github",
        },
        Object {
          "name": "provider",
          "type": "submit",
          "value": "google",
        },
      ],
      "method": "POST",
    }
  }
>
  <ButtonGroup
    className="oidc"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "gap": "4px",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="ant-btn-group oidc"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "gap": "4px",
          "justifyContent": "center",
        }
      }
    >
      <Divider
        plain={true}
      >
        <div
          className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center ant-divider-plain"
          role="separator"
        >
          <span
            className="ant-divider-inner-text"
            style={Object {}}
          >
            Or Sign 
            in
             with Email
          </span>
        </div>
      </Divider>
    </div>
  </ButtonGroup>
</OIDC>
`;
